ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Ledger/SupportsMempool.hs

```diff
@@ -79,6 +80,16 @@ data WhetherToIntervene
     -- them for it.
     deriving (Show)
 
+-- | Whether to keep track of the diffs produced by applying the transactions.
+--
+-- When getting a mempool snapshot, we will revalidate all the
+-- transactions but we won't do anything useful with the resulting
+-- state. We can safely omit computing the differences in this case.
+data ComputeDiffs
+  = ComputeDiffs
```

Comment:
Comment saying that this option should be used with resyncing the mempool with an updated selection.